ingestion/src/metadata/orm_profiler/interfaces/sqa_profiler_interface.py

```diff
@@ -0,0 +1,313 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""
+Interfaces with database for all database engine
+supporting sqlalchemy abstraction layer
+"""
+
+from datetime import datetime
+from typing import Dict, List, Optional, Union
+
+from sqlalchemy import Column
+from sqlalchemy.engine.row import Row
+from sqlalchemy.orm import DeclarativeMeta, Session
+
+from metadata.generated.schema.entity.data.table import TableProfile
+from metadata.generated.schema.tests.basic import TestCaseResult
+from metadata.generated.schema.tests.columnTest import ColumnTestCase
+from metadata.generated.schema.tests.tableTest import TableTestCase
+from metadata.orm_profiler.metrics.registry import Metrics
+from metadata.orm_profiler.profiler.runner import QueryRunner
+from metadata.orm_profiler.profiler.sampler import Sampler
+from metadata.orm_profiler.validations.core import validation_enum_registry
+from metadata.utils.connections import (
+    create_and_bind_session,
+    get_connection,
+    test_connection,
+)
+from metadata.utils.constants import TEN_MIN
+from metadata.utils.logger import sqa_interface_registry_logger
+from metadata.utils.timeout import cls_timeout
+
+logger = sqa_interface_registry_logger()
+
+
+class SQAProfilerInterface:
```

Comment:
should this extend an abstract object? If we want to add another profiler how would we know the methods to implement?